File: cpp/0028.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        // var
        int h_len = haystack.length();
        int n_len = needle.length();
        
        if (h_len == n_len)
            return haystack == needle ? 0 : -1;
        else if (h_len < n_len)
            return -1;
        else
            return haystack.find(needle);
    }
};

File: cpp/0028.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(haystack) < len(needle):
            return -1
        elif len(haystack) == len(needle):
            return 0 if (haystack == needle) else -1 
        else:
            for i in range(0, len(haystack) - len(needle) + 1):
                if haystack[i:i + len(needle)] == needle:
                    return i
        return -1
